presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -154,6 +155,11 @@ public SystemSessionProperties(
                             return concurrency;
                         },
                         value -> value),
+                integerSessionProperty(
+                        LOCAL_EXCHANGE_CONCURRENCY,
+                        "Number of local parallel hash partitioners per local exchange",
+                        1,
```

Comment:
@KBP-TDC `task_concurrency=16` can be bottlenecked on on single hash partitioner, that splits incoming pages into 16 streams. That's why something like I proposed is required.

Indeed I have chosen default value "1" as a safest approach, that will not brake anything. From tests that I have performed I haven't found a query where values greater then 1 decrease performance. There are queries where there is no difference however and as far I understand this code, in such case additional local exchanges just increases system memory usage without improving performance.

@raghavsethi I wanted to first introduce this switch play with it in real world on different workloads and then try to pick better default value.